crates/ibc/src/core/ics04_channel/events.rs

```diff
@@ -1,719 +1,733 @@
+        attributes.append(&mut v.acknowledgement.try_into()?);
+        attributes.push(v.dst_connection_id.into());
+
         Ok(AbciEvent {
             type_str: IbcEventType::WriteAck.as_str().to_string(),
             attributes,
         })
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+#[derive(Debug)]
 pub struct AcknowledgePacket {
-    pub packet: Packet,
+    packet_data: PacketDataAttribute,
+    timeout_height: TimeoutHeightAttribute,
+    timeout_timestamp: TimeoutTimestampAttribute,
+    sequence: SequenceAttribute,
+    src_port_id: SrcPortIdAttribute,
+    src_channel_id: SrcChannelIdAttribute,
+    dst_port_id: DstPortIdAttribute,
+    dst_channel_id: DstChannelIdAttribute,
+    channel_ordering: ChannelOrderingAttribute,
+    src_connection_id: PacketConnectionIdAttribute,
 }
 
 impl AcknowledgePacket {
-    pub fn src_port_id(&self) -> &PortId {
-        &self.packet.source_port
-    }
-    pub fn src_channel_id(&self) -> &ChannelId {
-        &self.packet.source_channel
-    }
-}
-
-impl From<AcknowledgePacket> for IbcEvent {
-    fn from(v: AcknowledgePacket) -> Self {
-        IbcEvent::AcknowledgePacket(v)
+    pub fn new(packet: Packet, channel_ordering: Order, src_connection_id: ConnectionId) -> Self {
+        Self {
+            packet_data: packet.data.into(),
+            timeout_height: packet.timeout_height.into(),
+            timeout_timestamp: packet.timeout_timestamp.into(),
+            sequence: packet.sequence.into(),
+            src_port_id: packet.source_port.into(),
+            src_channel_id: packet.source_channel.into(),
+            dst_port_id: packet.destination_port.into(),
+            dst_channel_id: packet.destination_channel.into(),
+            channel_ordering: channel_ordering.into(),
+            src_connection_id: src_connection_id.into(),
+        }
     }
 }
 
 impl TryFrom<AcknowledgePacket> for AbciEvent {
     type Error = Error;
 
     fn try_from(v: AcknowledgePacket) -> Result<Self, Self::Error> {
-        let attributes = Vec::<Tag>::try_from(v.packet)?;
+        let mut attributes = Vec::with_capacity(11);
+        attributes.append(&mut v.packet_data.try_into()?);
+        attributes.push(v.timeout_height.into());
+        attributes.push(v.timeout_timestamp.into());
+        attributes.push(v.sequence.into());
+        attributes.push(v.src_port_id.into());
+        attributes.push(v.src_channel_id.into());
+        attributes.push(v.dst_port_id.into());
+        attributes.push(v.dst_channel_id.into());
+        attributes.push(v.channel_ordering.into());
+        attributes.push(v.src_connection_id.into());
+
         Ok(AbciEvent {
             type_str: IbcEventType::AckPacket.as_str().to_string(),
             attributes,
         })
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+#[derive(Debug)]
 pub struct TimeoutPacket {
-    pub packet: Packet,
+    packet_data: PacketDataAttribute,
```

Comment:
Seems like ibc-go's `TimeoutPacket` doesn;t have the `packet_data` and `connection_id` fields. 
https://github.com/cosmos/ibc-go/blob/a0e59b8e7a2e1305b7b168962e20516ca8c98fad/modules/core/04-channel/keeper/events.go#L236-L254

Reply:
[cac17b4](https://github.com/cosmos/ibc-rs/pull/198/commits/cac17b477f00e46756d1e9ae43a5fb88466f93d1)